src/Components/Blazor/Blazor/src/Services/WebAssemblyUriHelper.cs

```diff
@@ -40,8 +40,7 @@ protected override void InitializeState()
             // client-side (Mono) use, so it's OK to rely on synchronicity here.
             var baseUri = WebAssemblyJSRuntime.Instance.Invoke<string>(Interop.GetBaseUri);
             var uri = WebAssemblyJSRuntime.Instance.Invoke<string>(Interop.GetLocationHref);
-            SetAbsoluteBaseUri(baseUri);
-            SetAbsoluteUri(uri);
+            base.InitializeState(uri, baseUri);
```

Comment:
Between you and @pranavkm this seems to change in every PR. Are we converging on a design here? I feel like this might need to have more parts than it currently does.

Reply:
We are meeting tomorrow (he can't do so today)